sdk/core/dma/dma_compartment.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * A function below claims the source and target addresses of the DMA interface.
+ * While, DMA is in progress, these addresses should not be freed
+ */
+__cheri_compartment("dma") int claim_dma(uint32_t *sourceAddress, 
```

Comment:
I think these (claim and free) functions are not what you mean to be exporting from the compartment.  I was expecting something something more like `write_conf` (though probably a fusion of `write_conf`, `write_strides`, and `start`), I think.

To support multiple users of DMA, we probably want that "`memmove_but_with_dma`" call, whatever it's named, to take a `Timeout` for waiting on the DMA engine to become available and for it to use a futex to communicate with the engine's IRQ handler?  But that's perhaps a discussion for later.